clippy_lints/src/use_self.rs

```diff
@@ -125,7 +125,7 @@ fn check_trait_method_impl_decl<'a, 'tcx>(
     let trait_method = cx
         .tcx
         .associated_items(impl_trait_ref.def_id)
-        .iter()
+        .in_definition_order()
         .find(|assoc_item| {
```

Comment:
this should become `find_by_name_and_kind(impl_item.ident, ty::AssocKind::Method, impl_trait_ref.def_id)`

Reply:
Fixed. Thanks!